src/checkers/rustchecker.py

```diff
@@ -0,0 +1,39 @@
+import toml
+import logging
+import urllib.request
+
+from src.lib import utils
+from src.lib.externaldata import Checker
+
+log = logging.getLogger(__name__)
+
+
+class RustChecker(Checker):
+    @staticmethod
+    def _should_check(external_data):
+        return external_data.checker_data.get("type") == "rust"
+
+    def check(self, external_data):
+        if not self._should_check(external_data):
+            log.debug("%s is not a rust type ext data", external_data.filename)
+            return
+
+        channel = external_data.checker_data.get("channel", "stable")
+        package_name = external_data.checker_data["package"]
+        target_name = external_data.checker_data["target"]
+
+        url = f"https://static.rust-lang.org/dist/channel-rust-{channel}.toml"
+
+        log.debug("Getting extra data info from %s; may take a while", url)
+        with urllib.request.urlopen(url) as response:
+            data = toml.loads(response.read().decode())
+
+        package = data["pkg"][package_name]
+        target = package["target"][target_name]
+        if target["available"]:
+            new_version = utils.get_extra_data_info_from_head(target["xz_url"])
+            new_version = new_version._replace(
+                checksum=target["xz_hash"], version=package["version"]
```

Comment:
`version` here looks something like `1.48.0 (7eac88abb 2020-11-16)`. I believe it is invalid for appstream. Maybe parse it with regex or else, and also extract date from it?

Reply:
is there any documentation on what version numbers are valid?

Reply:
I don't know any appstream documentation other than [this](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-releases), and it doesn't seem to specify any constraints for version numbers, so maybe I'm wrong and it allows any string as a version. Still we may want to parse it in order to get upstream release date.

Reply:
that’s not actually the release date, it the commit date of the commit used for the release.
For example rust `1.48.0` was released on `2020-11-19` but the version looks like this `1.48.0 (7eac88abb 2020-11-16)`
See: https://github.com/rust-lang/rust/releases/tag/1.48.0

Reply:
I think it's fine to strip the extra info like commit or build date.

Reply:
I agree that its fine for stable releases but different nightlies would then look the same for example `0.51.0-nightly`.

So we can either stick to the version strings that upstream provides or strip the extra info only on beta and stable.

Reply:
>different nightlies would then look the same for example `0.51.0-nightly`

I'd say it's fine. Unlikely that someone would case about exact commit.

Reply:
added code to remove everything after and including the first `(`